cloudstate/discovery_servicer.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 Lightbend Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import platform
+from dataclasses import dataclass
+from pprint import pprint
+from typing import List
+
+from google.protobuf.descriptor_pb2 import FileDescriptorSet, FileDescriptorProto
+from google.protobuf.descriptor_pool import Default
+
+from cloudstate import entity_pb2
+from cloudstate.entity_pb2_grpc import EntityDiscoveryServicer
+from cloudstate.event_sourced_entity import EventSourcedEntity
+
+
+@dataclass
+class CloudStateEntityDiscoveryServicer(EntityDiscoveryServicer):
+    event_sourced_entities: List[EventSourcedEntity]
+
+    def discover(self, request, context):
+        pprint(request)
+        descriptor_set = FileDescriptorSet()
+        for entity in self.event_sourced_entities:
+            for descriptor in entity.file_descriptors:
+                descriptor_set.file.append(FileDescriptorProto.FromString(descriptor.serialized_pb))
+        descriptor_set.file.append(
+            FileDescriptorProto.FromString(Default().FindFileByName('google/protobuf/empty.proto').serialized_pb)
+        )
+        descriptor_set.file.append(
+            FileDescriptorProto.FromString(Default().FindFileByName('cloudstate/entity_key.proto').serialized_pb)
+        )
+        descriptor_set.file.append(
+            FileDescriptorProto.FromString(Default().FindFileByName('google/protobuf/descriptor.proto').serialized_pb)
+        )
+        descriptor_set.file.append(
+            FileDescriptorProto.FromString(Default().FindFileByName('google/api/annotations.proto').serialized_pb)
+        )
+        descriptor_set.file.append(
+            FileDescriptorProto.FromString(Default().FindFileByName('google/api/http.proto').serialized_pb)
+        )
+        spec = entity_pb2.EntitySpec(
+            service_info=entity_pb2.ServiceInfo(
+                service_version='0.1.0',
+                service_runtime='Python ' + platform.python_version() + ' [' + platform.python_implementation() + ' ' +
+                                platform.python_compiler() + ']',
+                support_library_name='cloudstate-python-support',
+                support_library_version='0.1.0'
```

Comment:
I think we should get the version in an externalized way (get it from a file maybe or via setuptools). Maybe using pkg_resources

Reply:
Agree.